hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/datanode/fsdataset/impl/TestFsVolumeList.java

```diff
@@ -202,6 +205,14 @@ public void testDfsReservedForDifferentStorageTypes() throws IOException {
         .setConf(conf)
         .build();
     assertEquals("", 100L, volume4.getReserved());
+    FsVolumeImpl volume5 = new FsVolumeImplBuilder().setDataset(dataset)
+        .setStorageDirectory(
+            new StorageDirectory(
+                StorageLocation.parse("[NVDIMM]"+volDir.getPath())))
+        .setStorageID("storage-id")
+        .setConf(conf)
+        .build();
+    assertEquals("", 3L, volume5.getReserved());
```

Comment:
nit: Let's use better assertion statement `assertEquals(3L, volume5.getReserved());`

Reply:
In order to be consistent with the original code, the `assertEquals()`  here has three parameters, such as, lines  196 and 204 of the original code.

Reply:
Usually we can, but following original code style here is bad. When it fails, the original code gives up empty string. My code shows your expected value and actual value so you can debug. Please change it.

We can also file a JIRA to update all such cases where `assertEquals` can be improved.